geode-core/src/test/java/org/apache/geode/internal/cache/PartitionedRegionQueryEvaluatorTest.java

```diff
@@ -271,51 +272,148 @@ public void verifyPrimaryBucketNodesAreRetrievedForCqQuery() throws Exception {
   @Test
   public void verifyPrimaryBucketNodesAreRetrievedForCqQueryWithRetry() throws Exception {
     List<Integer> bucketList = createBucketList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
-    Set bucketSet = new HashSet<>(bucketList);
+    int bucket1 = 1;
+    Set<Integer> bucketSet = new HashSet<>(bucketList);
     when(query.isCqQuery()).thenReturn(true);
     PartitionedRegionQueryEvaluator prqe =
         spy(new PartitionedRegionQueryEvaluator(system, pr, query, mock(ExecutionContext.class),
             null, new LinkedResultSet(), bucketSet));
 
     for (Integer bid : bucketList) {
-      if (bid == 1) {
-        when(prqe.getPrimaryBucketOwner(bid)).thenReturn(null).thenReturn(remoteNodeA);
+      if (bid == bucket1) {
+        doReturn(null).doReturn(remoteNodeA).when(prqe).getPrimaryBucketOwner(bid);
       } else {
-        when(prqe.getPrimaryBucketOwner(bid)).thenReturn(remoteNodeA);
+        doReturn(remoteNodeA).when(prqe).getPrimaryBucketOwner(bid);
       }
     }
 
     Map<InternalDistributedMember, List<Integer>> bnMap = prqe.buildNodeToBucketMap();
 
-    assertThat(bnMap.size()).isEqualTo(1);
+    assertThat(bnMap.size()).isEqualTo(bucket1);
     assertThat(bnMap.get(remoteNodeA).size()).isEqualTo(10);
-    // Called 3 times : 2 times in retry and once for setting the return value
-    verify(prqe, times(3)).getPrimaryBucketOwner(1);
+    verify(prqe, times(2)).getPrimaryBucketOwner(1);
```

Comment:
I think that the reference to `bucket1` should be in the `verify()` method, not in the map size assertion, since we want to verify that we're calling `getPrinmaryBucketOwner()` multiple times specifically on the bucketId that returns null.